Sources/hyper-focus/action_handler.swift

```diff
@@ -41,27 +41,42 @@ enum ActionHandler {
             return false
         }
 
-        // add 'www.' to all block_hosts entries, this is not something users want to do manually
         let blockHosts = data.configuration.block_hosts
-        let blockHostsWithWWW = blockHosts.map { "www.\($0)" }
 
-        if blockHosts.contains(host) || blockHostsWithWWW.contains(host) {
+        if match(host, blockHosts) {
             error("blocked host, redirecting browser to block page")
             blockTab(data.activeTab)
             return true
         }
 
         debug("checking urls for any blocked matches")
 
+        let blockUrls = data.configuration.block_urls
         // the urls in the config are expected to have less params, so they are considered the subset
-        if data.configuration.block_urls.count > 0, data.configuration.block_urls.contains(where: { isSubsetOfUrl(supersetUrlString: url, subsetUrlString: $0) }) {
+        if blockUrls.count > 0 && blockUrls.contains(where: { isSubsetOfUrl(supersetUrlString: url, subsetUrlString: $0) }) {
             error("blocked url, redirecting browser to block page")
             blockTab(data.activeTab)
             return true
         }
 
         return false
     }
+    
+    static func match(_ str: String, _ regexpArray: [String]) -> Bool {
+        if regexpArray.count == 0 {
+            return false
+        }
+        if let matchedPattern = regexpArray.first(where: { pattern in
+            let regex = try! NSRegularExpression(pattern: pattern)
```

Comment:
I see your points, hadn't thought about them.

Triggering regex when the string has some sort of flag (like `//`) sounds like the best way to avoid the problems, as the  only alternative I can think of is having a `"regex"` setting in the config that turns regex matching on, and is off by default to avoid breaking old configs.

Should also be easy enough to implement, so unless you have a better solution (in which case, let me know!), I'll do that straight away.